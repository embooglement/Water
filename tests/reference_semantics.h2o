let test_null = func() {
	let x = null;
	let y = x;
	println(reference_equals(x, y));
	println(reference_equals(x, null));
};

let test_number = func() {
	let x = 0;
	let y = x;
	println(reference_equals(x, y));
};

let test_string = func() {
	let x = "Hello";
	let y = x;
	println(reference_equals(x, y));
};

let test_boolean = func() {
	let x = true;
	let y = false;
	println(reference_equals(x, y));
};

let test_function = func() {
	let x = func() {};
	let y = x;
	println(reference_equals(x, y));
};

let test_array = func() {
	let x = [];
	let y = x;
	println(reference_equals(x, y));

	y.push(42);
	println(x);
};

test_null();
test_number();
test_string();
test_boolean();
test_function();
test_array();
