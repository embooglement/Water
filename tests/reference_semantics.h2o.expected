true
true
false
true
false
true
true
[42]

